tools/wasm_chrome/branch_selector.mjs

```diff
@@ -106,21 +106,26 @@ class BranchSelector extends HTMLElement {
     return data;
   }
 
-  render() {
+  async render() {
     const urlParams = new URLSearchParams(window.location.search);
-    const selectedBranch = urlParams.get("branch") || DEFAULT_SELECTED_BRANCH;
+    let selectedBranch = urlParams.get("branch") || DEFAULT_SELECTED_BRANCH;
+    const branches = await this.getData();
+    // If the branch in the param does not exit, roll back to main :)
+    if(!branches.some(branch => branch.name == selectedBranch)){
```

Comment:
ignorable nit: This could just be a `find` instead. The only difference is that it stops as soon as it finds an occurance.